rust/routee-compass-powertrain/src/routee/vehicles/default/plug_in_hybrid.rs

```diff
@@ -0,0 +1,365 @@
+    updated_state.push(new_battery_energy.into());
+
+    updated_state
+}
+fn get_electrical_energy_from_state(state: &[StateVar]) -> Energy {
+    Energy::new(state[0].0)
+}
+
+fn get_gasoline_energy_from_state(state: &[StateVar]) -> Energy {
+    Energy::new(state[1].0)
+}
+
+fn get_remaining_battery_energy_from_state(state: &[StateVar]) -> Energy {
+    Energy::new(state[2].0)
+}
+
+fn get_battery_soc_percent(vehicle: &PlugInHybrid, state: &[StateVar]) -> f64 {
+    let battery_energy_unit = vehicle.battery_energy_unit;
+
+    let battery_capacity_kwh =
+        battery_energy_unit.convert(vehicle.battery_capacity, EnergyUnit::KilowattHours);
+
+    let remaining_battery_energy = get_remaining_battery_energy_from_state(state);
+
+    let remaining_battery_energy_kwh =
+        battery_energy_unit.convert(remaining_battery_energy, EnergyUnit::KilowattHours);
+
+    (remaining_battery_energy_kwh.as_f64() / battery_capacity_kwh.as_f64()) * 100.0
+}
+
+/// Compute the energy for the PHEV by converting gasoline to kWh.
+/// This uses a simplified operation in which we assume that if the battery
+/// SOC is greater than zero we can just operate on battery to traverse a link.
+/// This is not entirely realistic as it's possible to arrive at a link with
+/// 0.001% SOC and still need to use gasoline to traverse the link.
+///
+/// In the future we could make this more sophisticated by calculating
+/// the energy required to traverse the link using the battery and then
+/// finding the point at which we would have to switch to gasoline
+///
+/// Returns a tuple of (electrical_energy, electrical_energy_unit, gasoline_energy, gasoline_energy_unit)
+fn get_phev_energy(
+    vehicle: &PlugInHybrid,
+    battery_soc_percent: f64,
+    speed: (Speed, SpeedUnit),
+    grade: (Grade, GradeUnit),
+    distance: (Distance, DistanceUnit),
+) -> Result<(Energy, EnergyUnit, Energy, EnergyUnit), TraversalModelError> {
+    let electrical_energy_unit = vehicle
+        .charge_deplete_model
+        .energy_rate_unit
+        .associated_energy_unit();
+    let gasoline_energy_unit = vehicle
+        .charge_sustain_model
+        .energy_rate_unit
+        .associated_energy_unit();
+
+    if battery_soc_percent > 0.0 {
+        // assume we can just use the battery
+        let (electrical_energy, electrical_energy_unit) = vehicle
+            .charge_deplete_model
+            .predict(speed, grade, distance)?;
+        Ok((
+            electrical_energy,
+            electrical_energy_unit,
+            Energy::new(0.0),
+            gasoline_energy_unit,
+        ))
+    } else {
+        // just use the gasoline engine
+        let (gasoline_energy, gasoline_energy_unit) = vehicle
+            .charge_sustain_model
+            .predict(speed, grade, distance)?;
+        Ok((
+            Energy::new(0.0),
+            electrical_energy_unit,
+            gasoline_energy,
+            gasoline_energy_unit,
+        ))
+    }
+}
+
+#[cfg(test)]
```

Comment:
yeah it's tricky since the vehicle model might not return the _exact_ energy each time (especially if we swap it out with a new model) but perhaps we can add some "order of magnitude" test to make sure it's in a reasonable range

Reply:
got it. well, i'd follow your lead if this is a requirement for this. i think, maybe knowing it's in the right ballpark could give us a little reassurance our units are correct but then again we don't want to get into the business of writing tests here that are really routee-powertrain tests (tests of the underlying model).